src/llms/vector.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Vector search API.
+ *
+ * This module can be used to interact with the vector database configured
+ * in the Grafana LLM app plugin. That plugin must be installed, enabled and configured
+ * in order for these functions to work.
+ *
+ * The {@link enabled} function can be used to check if the plugin is enabled and configured.
+ */
+
+import { FetchError, getBackendSrv, logDebug } from "@grafana/runtime";
+import { LLM_PLUGIN_ROUTE } from "./constants";
+
+interface SearchResultPayload extends Record<string, any> { }
+
+/**
+ * A request to search for resources in the vector database.
+ **/
+export interface SearchRequest {
+  /**
+   * The name of the collection to search in.
+   **/
+  collection: string;
```

Comment:
naming nit: `collection` -> `collection_name`?

Reply:
I'm a bit soso on this one, I can't see what else it's gonna be other than a name 😁